src/casp/services/api/services/consensus_engine/strategies.py

```diff
@@ -0,0 +1,230 @@
+
+        :param query_ids: List of query cell IDs from the kNN query.
+        :param knn_query: The result of the kNN query.
+
+        :return: A list of summarized query neighbor context annotations.
+        """
+        ...
+
+
+class ConsensusStrategyType(str, Enum):
+    """
+    Enum representing the available methods for consensus engine calculations. Methods include:
+
+    - ``ONTOLOGY_AWARE``: Utilizes cell type ontology to enrich context summarization.
+    - ``CELL_TYPE_STATISTICS``: Counts cell types and distance statistics within the neighborhood of a query to
+        summarize context.
+    """
+
+    ONTOLOGY_AWARE = "ontology_aware"
+    CELL_TYPE_STATISTICS = "cell_type_statistics"
+
+
+class CellTypeStatisticsConsensusStrategy(ConsensusStrategyProtocol):
+    """
+    Handle cell type count consensus strategy, summarizing query neighbor context by cell type distribution.
+    """
+
+    def __init__(
+        self, cell_operations_dm: CellOperationsDataManager, cas_model: models.CASModel, include_dev_metadata: bool
+    ):
+        self.cell_operations_dm = cell_operations_dm
+        self.cas_model = cas_model
+        self.include_dev_metadata = include_dev_metadata
+
+    def summarize(self, query_ids: t.List[str], knn_query: MatchResult) -> schemas.QueryAnnotationCellTypeCountType:
+        """
+        Summarize query neighbor context by cell type distribution, querying a database for the distribution.
+
+        :param query_ids: List of query cell IDs.
+        :param knn_query: The result of the kNN query.
+
+        :return: An instance of `schemas.QueryAnnotationCellTypeCountType`, representing the summarized cell type
+            distribution of query neighbors.
+        """
+        temp_table_fqn = self.cell_operations_dm.insert_matches_to_temp_table(
+            query_ids=query_ids, knn_response=knn_query
+        )
+
+        if self.include_dev_metadata:
+            return self.cell_operations_dm.get_neighborhood_distance_summary_dev_details(
+                cas_model=self.cas_model, match_temp_table_fqn=temp_table_fqn
+            )
+
+        return self.cell_operations_dm.get_neighborhood_distance_summary(
+            cas_model=self.cas_model, match_temp_table_fqn=temp_table_fqn
+        )
+
+
+class CellOntologyResource:
+    """
+    Handles cell ontology resource data, such as the ancestors dictionary and cell ontology term ID mappings.
+
+    Attributes:
+        ancestors_dictionary: Maps cell ontology term IDs to their ancestor IDs.
+        ontology_term_id_to_name_dict: Maps cell ontology term IDs to cell type names.
+
+    Raises:
+        ValueError: If either ``ancestors_dictionary`` or ``cell_ontology_term_id_to_cell_type`` is missing from the
+            provided dictionary.
+    """
+
+    def __init__(self, cell_ontology_resource_dict: t.Optional[t.Dict[str, t.Any]] = None):
```

Comment:
include cell ontology version that is used, and a permalink to where the ontology obo file is located.